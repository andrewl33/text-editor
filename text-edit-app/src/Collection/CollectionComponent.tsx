import * as React from "react";
import { Container, Grid } from "semantic-ui-react";
import { ListComponent } from "../generic/Lists/ListComponent";
import SidebarComponent from "../generic/Sidebar/SidebarComponent";
import { CollectionComponentProps } from "../types";
// TODO: add styling
export const CollectionComponent = (props: CollectionComponentProps) => {
  const { name, createDate, isLocked, users } = props;

  return (
    <Grid style={{ marginTop: 0, height: "100%" }}>
      <Grid.Column width={13} style={{ paddingRight: 0, paddingTop: 0 }}>
        <ListComponent header="Files" {...props} />
      </Grid.Column>
      <Grid.Column
        width={3}
        textAlign="center"
        style={{
          paddingRight: "2em",
          backgroundColor: "#3180d4",
          height: "100%"
        }}
      >
        <Container>
          <SidebarComponent
            name={name}
            pageType="collection"
            createDate={createDate}
            isPrivate={isLocked}
            users={users}
            // tslint:disable-next-line
            onNameChange={a => {}}
          />
        </Container>
      </Grid.Column>
    </Grid>
  );
};
