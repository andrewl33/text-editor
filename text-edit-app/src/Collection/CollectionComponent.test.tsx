import * as React from "react";
import * as renderer from "react-test-renderer";
import { CollectionComponent } from "./CollectionComponent";

const items = [
  {
    id: "1",
    name: "name1",
    tags: [],
    date: "1-1-2001"
  }
];

const props = {
  name: "test",
  createDate: "1-1-2001",
  isLocked: false,
  users: ["user1", "user2"],
  onNameChange: jest.fn(),
  onAddUser: jest.fn(),
  onRemoveUser: jest.fn(),
  onAddFile: jest.fn(),
  items,
  onRemoveFile: jest.fn(),
  onFileClick: jest.fn(),
  accountName: ""
};

describe("CollectionComponent", () => {
  test("renders correctly", () => {
    const component = renderer.create(<CollectionComponent {...props} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
