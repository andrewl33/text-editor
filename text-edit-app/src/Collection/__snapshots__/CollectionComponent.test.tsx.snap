// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionComponent renders correctly 1`] = `
<div
  className="ui grid"
  style={
    Object {
      "height": "100%",
      "marginTop": 0,
    }
  }
>
  <div
    className="thirteen wide column"
    style={
      Object {
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <div>
      <table
        className="ui blue large celled inverted middle aligned table"
        divided="true"
        relaxed="true"
        style={
          Object {
            "borderRadius": 0,
            "color": "rgba(255,255,255,.9)",
          }
        }
      >
        <thead
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
              colSpan="2"
            >
              Files
            </th>
            <th
              className=""
            >
              Tags
            </th>
            <th
              className=""
              colSpan="2"
            >
              Dates
            </th>
            <th
              className=""
              colSpan="2"
            >
              Delete
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        >
          <tr
            className=""
          >
            <td
              className="selectable"
              colSpan="2"
              onClick={[Function]}
              style={
                Object {
                  "padding": ".92857143em .78571429em",
                }
              }
            >
              name1
            </td>
            <td
              className=""
            />
            <td
              className=""
              colSpan={2}
            >
              1-1-2001
            </td>
            <td
              className=""
              colSpan={2}
            >
              <button
                className="ui button"
                onClick={[Function]}
                role="button"
              >
                <i
                  aria-hidden="true"
                  className="close icon"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="ui input"
      >
        <input
          onChange={[Function]}
          placeholder="File url..."
          type="text"
          value=""
        />
        <button
          className="ui button"
          onClick={[Function]}
          role="button"
          type="submit"
        >
          Add
        </button>
      </div>
    </div>
  </div>
  <div
    className="center aligned three wide column"
    style={
      Object {
        "backgroundColor": "#3180d4",
        "height": "100%",
        "paddingRight": "2em",
      }
    }
  >
    <div
      className="ui container"
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Name
          </div>
          <span
            className="name"
          >
            test
            <i
              aria-hidden="true"
              className="edit link icon"
              onClick={[Function]}
            />
          </span>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Date Created
          </div>
          1-1-2001
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Private
          </div>
          No
        </div>
        <div
          className="item users"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Users
          </div>
          <div
            className="ui label"
            onClick={[Function]}
          >
            user1
          </div>
          <div
            className="ui label"
            onClick={[Function]}
          >
            user2
          </div>
          
        </div>
      </div>
    </div>
  </div>
</div>
`;
